addon/lib/StudyTelemetryCollector.jsm

```diff
@@ -33,47 +32,50 @@ XPCOMUtils.defineLazyModuleGetter(this, "PlacesUtils",
 XPCOMUtils.defineLazyServiceGetter(this, "Telemetry",
   "@mozilla.org/base/telemetry;1", "nsITelemetry");
 
+// unit-tested study helpers
+XPCOMUtils.defineLazyModuleGetter(
+  this, "Helpers", "resource://esper-pioneer-shield-study/lib/Helpers.jsm"
+);
+
 /**
  * Note: Setting studyUtils in this constructor due to above comments
  */
 class StudyTelemetryCollector {
 
-  constructor(studyUtils, variation) {
-    this.variation = variation;
-    this.studyUtils = studyUtils;
+  constructor(pioneerUtils) {
+    this.pioneerUtils = pioneerUtils;
   }
 
   async start() {
 
-    this.telemetry({ event: "esper-init" });
+    console.log('ESPER Study init. Will now await full telemetry initialization before collecting the payload for this study.');
 
     // Ensure that we collect telemetry payloads only after it is fully initialized
     // See http://searchfox.org/mozilla-central/rev/423b2522c48e1d654e30ffc337164d677f934ec3/toolkit/components/telemetry/TelemetryController.jsm#295
-    TelemetryController.promiseInitialized().then(() => {
-
-      try {
+    await TelemetryController.promiseInitialized();
 
-        this.collectAndSendTelemetry();
+    try {
 
-      } catch (ex) {
-        // TODO: how are errors during study execution reported?
-        // this.studyUtils.telemetryError();
-        Components.utils.reportError(ex);
-      }
+      this.collectAndSendTelemetry();
 
-    });
+    } catch (ex) {
+      // TODO: how are errors during study execution reported?
+      // this.pioneerUtils.telemetryError();
+      Components.utils.reportError(ex);
+    }
 
   }
 
-  telemetry(payload) {
-    if (StudyTelemetryCollector.allowedToSendTelemetry()) {
-      this.studyUtils.telemetry(payload);
+  async telemetry(schemaName, schemaVersion, payload) {
+    if (await this.allowedToSendTelemetry()) {
+      await this.pioneerUtils.submitEncryptedPing(schemaName, schemaVersion, payload);
+      console.log('ESPER Telemetry sent (encrypted)', JSON.stringify(payload));
     } else {
-      console.log('ESPER telemetry not sent due to privacy preferences', payload);
+      console.log('ESPER Telemetry not sent due to privacy preferences', JSON.stringify(payload));
     }
   }
 
-  static allowedToSendTelemetry() {
+  async allowedToSendTelemetry() {
```

Comment:
I don't think you need this as it is already handled by pioneer-utils.

Reply:
True. This was implemented in response to https://github.com/motin/esper-pioneer-shield-study/issues/8 at a time when neither shield-utils nor pioneer-utils did this check properly. Since it now does, this can be removed :)